eggplant/invitations/utils.py

```diff
@@ -1,15 +1,15 @@
-from uuid import uuid4
-
 from django.contrib.auth import get_user_model
 
 from allauth.account.models import EmailAddress, EmailConfirmation
 
 
 def create_verified_user(invitation):
-    password = uuid4().hex
     User = get_user_model()
-    user = User.objects.create_user(invitation.email, invitation.email,
-                                    password)
+    # Instead of setting the users password to some UUID hex string.
+    # Calling create_user without a password whis will call
+    # set_unusable_password() and enable the use of has_usable_password() on
+    # the user object.
+    user = User.objects.create_user(invitation.email, invitation.email)
```

Comment:
I don't think there is any harm in setting it to be a UUID string but you are absolutely right, we should use the existing django API. I can remember the flow of the invitation now but I think I didn't want the password to be empty - "unusable" so I could use `PasswordResetForm` or perhaps there was something else like authn backend that I wanted to use from the invitation authn backend... I think the idea was that user accepts invitation - the system logs him in, he updates the profile and resets the password... I don't know if I ever got to the point of resetting the password though. I don't know if this is because the flow has changed or we didn't have user story for this...
Anyhow, I hope this helps.